syntax = "proto3";

package customer;

message Customer{
  string customer_ref = 1;
  string name = 2;
  string email = 3;
}

message CustomerRegistrationRequest{
  Customer customer = 1;
}

message CustomerRegistrationResponse{
  bool success = 1;
  string message = 2;
}

message GetCustomerByReferenceRequest{
  string customer_ref = 1;
}

message GetCustomerByReferenceResponse{
  Customer customer = 1;
  bool found = 2;
}

message GetCustomerByEmailRequest{
  string email = 1;
}

message GetCustomerByEmailResponse{
  Customer customer = 1;
  bool found = 2;
}

service CustomerRegistration{
  rpc register_new_customer (CustomerRegistrationRequest) returns (CustomerRegistrationResponse);
  rpc get_customer_by_reference(GetCustomerByReferenceRequest) returns (GetCustomerByReferenceResponse);
  rpc get_customer_by_email(GetCustomerByEmailRequest) returns (GetCustomerByEmailResponse);
}